src/ObjPropsDynamicSetter/ObjectExtensions.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+
+namespace ObjPropsDynamicSetter
+{
+    /// <summary>
+    /// Provides extension methods to access object properties via their names.
+    /// </summary>
+    public static class ObjectExtensions
+    {
+        private const char Delimiter = '.';
+
+        /// <summary>
+        /// Gets the property info via its name.
+        /// </summary>
+        /// <param name="obj">Object instance.</param>
+        /// <param name="name">Property name.</param>
+        /// <param name="includeNonPublic">Indicates whether to include non-public properties.</param>
+        /// <returns>Property information.</returns>
+        /// <exception cref="ArgumentNullException">Object is null.</exception>
+        /// <exception cref="ArgumentException">Property is not found in object or property name is null or empty.</exception>
+        public static PropertyInfo GetPropertyInfo(this object obj, string name, bool includeNonPublic = false)
+        {
+            ValidateParameters(obj, name);
+            return obj.GetPropertyDetails(GetPropertyPathItems(name), includeNonPublic).PropertyInfo;
+        }
+
+        /// <summary>
+        /// Gets the property value via its name.
+        /// </summary>
+        /// <typeparam name="T">Property type.</typeparam>
+        /// <param name="obj">Object instance.</param>
+        /// <param name="name">Property name.</param>
+        /// <param name="includeNonPublic">Indicates whether to include non-public properties.</param>
+        /// <returns>Property value.</returns>
+        /// <exception cref="ArgumentNullException">Object is null.</exception>
+        /// <exception cref="ArgumentException">Property is not found in object or property name is null or empty.</exception>
+        /// <exception cref="InvalidCastException">Property value cannot be casted to expected return type.</exception>
+        public static T GetPropertyValue<T>(this object obj, string name, bool includeNonPublic = false)
```

Comment:
Codacy found an issue: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/manual/Molnix888/obj-props-dynamic-setter/pullRequest?prid=6241271)